libraries/ts/src/margin/marginClient.ts

```diff
@@ -119,7 +123,7 @@ export class MarginClient {
     const instructions = ["repay", "borrow", "deposit", "withdraw"]
     let tradeAction = ""
     for (let i = 0; i < instructions.length; i++) {
-      if (transaction.meta?.logMessages?.some(log => log.toLowerCase().includes(instructions[i]))) {
+      if (transaction.meta?.logMessages?.some(tx => tx.toLowerCase().includes(instructions[i]))) {
         tradeAction = instructions[i]
```

Comment:
I think removing lowercase is possble and it's good to be as specific as possible
```suggestion
      if (transaction.meta?.logMessages?.some(logLine => logLine.includes(instructionLogHeaders[i]))) {
        tradeAction = instructionLogHeaders[i]
```